### PR TITLE
Jwoo/ios custom element update

### PR DESCRIPTION
**Pull request by nesalang**
- Resolves the issue #2220 
- it will break shared model unit test (I need to update the test).

### CODE REVIEW

File: samples/Tests/Action.CustomParsing.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+	"$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+	"type": "AdaptiveCard",
+	"version": "1.0",
+	"body": [
+		{
+			"type": "TextBlock",
+			"text": "This card has a custom action. If renderer can redner the custom type, it will render a button.",
```
Comment (khouzam) [inline samples/Tests/Action.CustomParsing.json]: >redner [](start = 59, length = 6)

render

File: source/dotnet/Test/AdaptiveCards.Test/AllPayloadTests.cs [hunk 2]
```diff
@@ -126,6 +126,7 @@ public void TestAllTestCards()
                     "FlightItinerary_es_fail",
                     "Action.DuplicateIds",
                     "Action.NestedDuplicateIds",
+                    "Action.CustomParsing",
```
Comment (khouzam) [inline source/dotnet/Test/AdaptiveCards.Test/AllPayloadTests.cs]: I don't think that this test should fail parsing.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRegistration.mm [hunk 3]
```diff
@@ -118,16 +126,21 @@ - (void)setActionSetRenderer:(id<ACRIBaseActionSetRenderer>)actionsetRenderer
     _actionSetRenderer = actionsetRenderer;
 }
 
-- (void) setActionRenderer:(ACRBaseActionElementRenderer *)renderer cardElementType:(NSNumber *)cardElementType
+- (void)setActionRenderer:(ACRBaseActionElementRenderer *)renderer cardElementType:(NSNumber *)cardElementType
 {
+    // custom action must be registered through set custom action renderer method
+    if(cardElementType.longValue > ACRUnknownAction) {
```
Comment (khouzam) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRegistration.mm]: >cardElementType.longValue > ACRUnknownAction [](start = 7, length = 44)

If this is a requirement of the enum, then at least put a comment in the enum that the order is important and that the ACRUnknownAction needs to always be last.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCustomRenderer.mm [hunk 4]
```diff
@@ -52,6 +52,7 @@ - (UIView *)render:(UIView<ACRIContentHoldingView> *)viewGroup
             [viewGroup addArrangedSubview:wrappingview];
         }
     }
+ */
```
Comment (khouzam) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCustomRenderer.mm]: Needs to be removed

**Comment by khouzam** [comment]
>     TEST_CLASS(CustomParsingForIOSTest)

Doesn't need to be IOS specific since this is the shared model

---
Refers to: source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModelUnitTest/CustomParsingForIOSTest.cpp:18 in 19d5c92. [](commit_id = 19d5c9267b4ca3eead5718ae93c06c444b271a0b, deletion_comment = False)

**Comment by nesalang** [comment]
replaced by #2325

**Pull request closed without merging**